src/app/content/hooks/locationChange.ts

```diff
@@ -1,60 +1,113 @@
 import css from 'cnx-recipes/styles/output/intro-business.json';
-import { RouteHookBody } from '../../navigation/types';
+import { BOOKS } from '../../../config';
+import { Match, RouteHookBody } from '../../navigation/types';
 import { AppServices, FirstArgumentType, MiddlewareAPI } from '../../types';
+import { assertDefined } from '../../utils';
 import { receiveBook, receivePage, requestBook, requestPage } from '../actions';
 import { content } from '../routes';
 import * as select from '../selectors';
 import { ArchivePage, Book } from '../types';
-import { flattenArchiveTree, getContentPageReferences } from '../utils';
+import { flattenArchiveTree, getContentPageReferences, getPageIdFromUrlParam, getUrlParamForPageId } from '../utils';
 
 const fontMatches = css.match(/"(https:\/\/fonts\.googleapis\.com\/css\?family=.*?)"/);
 const fonts = fontMatches ? fontMatches.slice(1) : [];
 
 const hookBody: RouteHookBody<typeof content> = (services) => async({match}) => {
-  const {dispatch, getState, fontCollector, archiveLoader} = services;
+  const {fontCollector} = services;
+
+  fonts.forEach((font) => fontCollector.add(font));
+
+  await resolveBook(services, match).then(resolvePage(services, match));
```

Comment:
Form a cursory skim, it seems like `.then` does not do anything... but `resolvePage(...)` returns a function rather than evaluating which was a bit unexpected. What about something like:

```suggestion
  const [book: Book, bookLoader: ReturnType<AppServices['archiveLoader']['book']>] = await resolveBook(services, match);
  await resolvePage(services, match, book, bookLoader);
```